src/systems/physics/Physics.cc

```diff
@@ -4165,6 +4179,97 @@ void PhysicsPrivate::UpdateCollisions(EntityComponentManager &_ecm)
       });
 }
 
+//////////////////////////////////////////////////
+void PhysicsPrivate::UpdateRayIntersections(EntityComponentManager &_ecm)
+{
+  GZ_PROFILE("PhysicsPrivate::UpdateRayIntersections");
+  // Quit early if the RaycastData component hasn't been created.
+  // This means there are no systems that need raycasting information
+  if (!_ecm.HasComponentType(components::RaycastData::typeId))
+    return;
+
+  // Assume that there is only one world entity
+  Entity worldEntity = _ecm.EntityByComponents(components::World());
+
+  if (!this->entityWorldMap.HasEntity(worldEntity))
+  {
+    gzwarn << "Failed to find world [" << worldEntity << "]." << std::endl;
+    return;
+  }
+
+  auto worldRayIntersectionFeature =
+      this->entityWorldMap.EntityCast<RayIntersectionFeatureList>(worldEntity);
+
+  if (!worldRayIntersectionFeature)
+  {
+    static bool informed{false};
+    if (!informed)
+    {
+      gzdbg << "Attempting process ray intersections, but the physics "
+             << "engine doesn't support ray intersection features. "
+             << "Ray intersections won't be computed."
+             << std::endl;
+      informed = true;
+    }
+    return;
+  }
+
+  // Go through each entity that has a RaycastData components, trace the
+  // rays and store the results
+  _ecm.Each<components::RaycastData,
+            components::Pose>(
+      [&](const Entity &_entity,
+          components::RaycastData *_raycastData,
+          components::Pose */*_pose*/) -> bool
+      {
+        // Retrieve the rays from the RaycastData component
+        const auto &rays = _raycastData->Data().rays;
+
+        // Clear the previous results
+        auto &results = _raycastData->Data().results;
+        results.clear();
```

Comment:
Add `results.resize(rays.size())` to avoid memory allocations in the for loop?

Reply:
I guess you mean `reserve`, due to it will work fine with push_back

Reply:
Good catch!

Done in https://github.com/gazebosim/gz-sim/pull/2514/commits/77802821ac38854e96542372371588a98b54dfe4

Reply:
> I guess you mean `reserve`, due to it will work fine with push_back

Yes, I did mean `reserve`. Thanks for the catch.